numba/cuda/tests/cudapy/test_frexp_ldexp.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import math
+from numba import cuda
+from numba.types import float32, float64, int32
+from numba.cuda.testing import unittest, CUDATestCase
+
+
+def simple_frexp(x, exp, arg):
+    x[0], exp[0] = math.frexp(arg)
+
+
+def simple_ldexp(val, arg, exp):
+    val[0] = math.ldexp(arg, exp)
+
+
+class TestCudaFrexpLdexp(CUDATestCase):
+    def template_test_frexp_ldexp(self, value, npyty, npmty):
```

Comment:
In my previous comment I'd only intended to convey that `test_frexp_f4` and `test_frexp_f8` should be combined into a single function, e.g. something along the lines of: 

```python
    def _test_frexp(self, nptype, nbtype):
        compiled = cuda.jit(void(nbtype[:], int32[:], nbtype))(simple_frexp)
        arg = 3.1415
        aryx = np.zeros(1, dtype=nptype)
        aryexp = np.zeros(1, dtype=np.int32)
        compiled[1, 1](aryx, aryexp, arg)
        np.testing.assert_array_equal(aryx, nptype(0.785375))
        self.assertEquals(aryexp, 2)

    def test_frexp_f4(self):
        self._test_frexp(np.float32, types.float32)

    def test_frexp_f8(self):
        self._test_frexp(np.float64, types.float64)
```

and making a function `_test_ldexp` for `test_ldexp_f4` and `test_ldexp_f8` to call, rather than combining the test of `ldexp` and `frexp` into a single function - apologies for any confusion.